tests/utils.rs

```diff
@@ -239,6 +239,43 @@ pub fn transfer_batch_internal(
     assert!(res.contains(&(from, codec)));
 }
 
+pub fn transform_internal(
+    mtk: &Program,
+    from: u64,
+    token_id: u128,
+    amount: u128,
+    accounts: Vec<ActorId>,
+    nft_ids: Vec<Vec<TokenId>>,
+    nfts_metadata: Vec<Vec<Option<TokenMetadata>>>,
+) {
+    let res = mtk.send(
+        from,
+        MyMTKAction::Transform {
+            id: token_id,
+            amount,
+            accounts,
+            nft_ids: nft_ids.clone(),
+            nfts_metadata,
+        },
+    );
+
+    let mut ids = vec![];
+    for ids_subset in nft_ids.iter() {
+        for id in ids_subset.iter() {
+            ids.push(*id);
+        }
+    }
```

Comment:
Can we precalculate vec capacity? And don't use `iter` explicitly if it is not required.
```suggestion
    let mut ids = Vec::with_capacity(nft_ids.len() * ids_subset.len());
    for ids_subset in &nft_ids {
        for id in &ids_subset {
            ids.push(*id);
        }
    }
```